src/main/java/com/google/step/model/User.java

```diff
@@ -37,4 +37,18 @@ public User(long id, String email, String username, String bio) {
     this.bio = bio;
     this.photoBlobKey = Optional.empty();
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null || !(obj instanceof User)) {
+      return false;
+    }
+    User user = (User) obj;
+    return user.id == this.id
+        && ((user.username == null && this.username == null) || user.username.equals(this.username))
+        && ((user.bio == null && this.bio == null) || user.bio.equals(this.bio))
+        && ((user.email == null && this.email == null) || user.email.equals(this.email))
+        && ((user.photoBlobKey == null && this.photoBlobKey == null)
+            || user.photoBlobKey.equals(this.photoBlobKey));
```

Comment:
Since these are all `String`s, you can simply do `user.username == this.username`. This will take care of the null equality check as well as the normal equality check.

Reply:
I think the == operator only compares reference but I want it to compare the content of the string.